.github/workflows/buildimages-update.yml

```diff
@@ -100,3 +104,43 @@ jobs:
           # the action fetches all branches and tags, in our case the branches we care about are already fetched
           # if they exist, so we can skip the fetch
           skip_fetch: true
+
+      - name: Check if PR exists
+        id: check_pr
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          base_branch=${{ github.ref }} # Get the full ref
+          base_branch=${base_branch#refs/heads/} # Remove 'refs/heads/' from the ref
+          prs=$(gh pr list \ # prs variable contains the number of PRs already created that match head and base branches
+              --repo "$GITHUB_REPOSITORY" \
+              --head ${{ inputs.branch }} \
+              --base $base_branch \
+              --json title \
+              --jq 'length')
+          if [ $prs -eq 0 ]; then
+            echo "CREATE_PR=true" >> $GITHUB_OUTPUT
+          fi 
+
+      # Create PR only if there is no pre-existing PR on the branch
+      - name: Create PR
+        if: ${{ steps.check_pr.outputs.CREATE_PR == 'true' }}
+        env:
+          TMP_PR_BODY_PATH: /tmp/pr_body
+          GH_TOKEN: ${{ github.token }}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Same question as above

Reply:
[Nevermind !](https://stackoverflow.com/a/66903734)